pkg/config/sync.go

```diff
@@ -104,6 +115,37 @@ func updatePublicConfig() {
 	Workspace = internalConfig.Workspace.DeepCopy()
 }
 
+func fillOpenShiftRouteSuffix(client crclient.Client) error {
+	if !infrastructure.IsOpenShift() {
+		return nil
+	}
+
+	testRoute := &routeV1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: configNamespace,
+			Name:      "devworkspace-controller-test-route",
+		},
+		Spec: routeV1.RouteSpec{
+			To: routeV1.RouteTargetReference{
+				Kind: "Service",
+				Name: "devworkspace-controller-test-route",
+			},
+		},
+	}
+
+	err := client.Create(context.TODO(), testRoute)
```

Comment:
on DevSanbox stage I faced the following failure:
![Screenshot_20210917_150634](https://user-images.githubusercontent.com/5887312/133991761-3da00b94-3b66-4bf5-a01a-772d43a16c32.png)
![Screenshot_20210917_150649](https://user-images.githubusercontent.com/5887312/133991765-1552b169-e6d4-42d6-b87f-42acf1e7b882.png)
So, I had to remove it manually.

:man_facepalming: for myself not to capture whole error message but it was about: `can't create test route since it already exists`. Not sure why terminal entered into that state but maybe we need first to check if it exists, and use if it does.

Reply:
I left that comment and in the next commit I see it's fixed. @amisevsk this kudo for you :+1: for addressing feedback before it's even flushed :-D
So, please just resolve this thread )

Reply:
There were a good number of edge cases in OpenShift handling, hopefully it's stable now.